src/routeGeneration/templateHelpers.ts

```diff
@@ -383,16 +394,27 @@ export class ValidationService {
       } else if (additionalProperties === false) {
         Object.keys(value).forEach((key: string) => {
           if (!alreadyValidatedProperties.has(key)) {
```

Comment:
I think there should be and extracted function getExcessPropertyForModel(model, properties) that returns an array of excess properties. That would be a lot easier to read.

Reply:
I created a new function called `isAnExcessProperty` that still utilizes the `Set` (for performance reasons) but should make the code more readable.